packages/cli/src/utils/replace-blitz-dependency.ts

```diff
@@ -0,0 +1,21 @@
+import {writeJSONSync} from 'fs-extra'
+import {join} from 'path'
+
+import {fetchDistTags} from './npm-fetch'
+
+export const replaceBlitzDependency = async (pkg: any, desinationPath: string, cliVersion: string) => {
+  const distTags = await fetchDistTags('blitz')
+
+  if (!distTags) {
+    return
+  }
+
+  // `blitz new` from a global install of `blitz@latest` will result in `"blitz": "latest"` in package.json
+  // `blitz new` from a global install of `blitz@canary` will result in `"blitz": "canary"` in package.json
+  // Otherwise, fall back on canary
+  const newVersion =
+    cliVersion === distTags.latest ? 'latest' : cliVersion === distTags.canary ? 'canary' : 'canary'
```

Comment:
I think this logic needs to be more loose and not check for an exact version match. So basically this:

```js
newVersion = cliVersion.includes('canary') ? 'canary' : 'latest'
```